modules/core/src/main/scala/org/tessellation/infrastructure/trust/TrustStorageUpdater.scala

```diff
@@ -0,0 +1,39 @@
+package org.tessellation.infrastructure.trust
+
+import cats.effect.std.Supervisor
+import cats.effect.{Async, Ref}
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.schema.SnapshotOrdinal
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.gossip.Gossip
+
+trait TrustStorageUpdater[F[_]] {
+  def update: F[Unit]
+}
+
+object TrustStorageUpdater {
```

Comment:
this looks a little bit odd. there is a `TrustStorage` but you cannot update it directly, because you want to execute an additional logic when updated, so you created `TrustStorageUpdater` that updates the storage and does that additional work. IMO, the storage should be self-containing and you should execute the update method on the storage BUT the storage can publish a stream with updates, so you can react on each update and execute gossip.spread or whatever action you want to take. see NodeStorage and NodeStateDaemon for example as it's exactly the same pattern. We update the node state and we want to gossip state changes as soon as the change happened

Reply:
So, the update occur at specified ordinals. The `next` ordinal trust storage is updated every `snapshotInterval` ordinals and the `current` ordinal trust storage is updated `snapshotInterval` + `delayToUpdateCurrent` ordinals. After `next` is updated, it's `SnapshotOrdinalPublicTrust` is gossiped out.

Instead of adding the additional dependencies to get the ordinal and gossip out the public trusts as appropriate, as well as the additional complexity of triggering the update at particular times, `TrustStorageUpdater` was created as a service that triggers every minute, calls the updates for the different ordinals, and gossips out the public trusts. The trusts only update when the ordinal thresholds have been crossed. Same for the gossip.

Reply:
@cngo-github
> Instead of adding the additional dependencies to get the TrustStorage and gossip out the public trusts as appropriate, as well as the additional complexity of triggering the update at particular times,

can you describe what you mean by `additional dependencies` and `additional complexity`? TrustStorage should do not care about gossiping or triggering updates. It should be external anyway.

Reply:
@marcinwadon I suggested a separate service to do the logic of when/how to update the trust store as well as gossip because that doesn't feel like it belongs in the store. IMO the trust store is already doing too much.

I agree with you that the logic should be self-contained, which is why I can see a simplification where the two classes we use (one `TrustMap` and two `OrdinalTrustMap`s) self-supporting and that trust store just calls into each of those things to do whatever changes required. I have to study the code again to make sure that suggestion makes sense but that's how I'd do it.

We can push the updater logic in there too, meaning we have to inject 2 more services in the `make` to accommodate that, and to me it feels like that logic doesn't belong there.

Reply:
@sdavidp 
> @marcinwadon I suggested a separate service to do the logic of when/how to update the trust store as well as gossip because that doesn't feel like it belongs in the store. IMO the trust store is already doing too much.

I 100% agree with that. What I mean is that it could be reactive instead of "long polling". So, it can react on the updates instead of polling the trust and verifying if something has changed.

Reply:
Like -> every time the next or current ordinal trust is updated, we want to gossip it. instead of checking every minute if anything has changed, we can react on updates and gossip it immediately after update (yet, it won't be a part of `TrustStore` to keep single responsibility principle)

Reply:
@marcinwadon I believe we discussed that but decided the polling would be simpler and not a drain on cpu or something like that.

Are you thinking we could push the data to be gossipped into a `SignallingRef` and instead of a polling daemon we gossip the object pulled from the `.discrete` stream?

cc: @cngo-github @ryle-goehausen

Reply:
@ryle-goehausen has asked that this changed be pushed to another PR. I made a ticket for this:

https://constellationnetwork.atlassian.net/browse/PROT-327

Reply:
Yes, this is not a priority to implement immediately as we can rely on infrequent small updates for now. Since this does not require substantial resources the daemon should solve the immediate problem without the added complexity of converting everything to reactive. At some point in the future if we need more reactive operations or they induce a heavier load (vs. just checking an ordinal value,) then we can implement the other ticket. The underlying logic should be relatively similar in both cases.